visualizer/src/response.rs

```diff
@@ -16,8 +18,30 @@ impl Display for ResponseFieldMask {
     }
 }
 
+impl TryFrom<String> for ResponseFieldMask {
+    type Error = anyhow::Error;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        match value.as_str() {
+            "svg" => Ok(ResponseFieldMask::Svg),
+            "png" => Ok(ResponseFieldMask::Png),
+            _ => Err(anyhow::anyhow!("invalid response filed mask: {}", value)),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct OutputMask(pub HashSet<ResponseFieldMask>);
+
+impl OutputMask {
+    pub fn contains(&self, key: &ResponseFieldMask) -> bool {
+        // empty output mask means that all fields must present
+        self.0.is_empty() || self.0.contains(key)
```

Comment:
I suggest to move that out of the business logic and implement inside servers instead. The fact that in cases of empty mask all fields should be returned was specified in API design guideline (https://google.aip.dev/157) and is more related to our GRPC and HTTP API design rather than the actual visualization logic. In case if empty mask would be provided by a user, the visualization-server should just create a mask with all possible fields.

Reply:
Actually, in that case, we may probably remove `OutputMask` structure completely, and use `HashSet<ResponseFieldMask>` instead, as `contains` function would be the same for `HashSet` and `OutputMask`. The only problem would be with getting all possible `ResponseFieldMask` options. We may either implement it for the `ResponseFieldMask` struct itself (e.g. `ResponseFieldMask::all() -> HashSet<ResponseFieldMask>`), or It may be a good idea to leave `OutputMask` and just manually implement `Default` for it with all possible `ResponseFieldMask` options

Reply:
I agree that we can leave OutputMask and inplement function with all possbile options, but I don't like call it `default`. my suggestion is to call it `fn full() -> Self`